### PR TITLE
repair: Ignore keyspace that is removed in sync_data_using_repair

### LINKED ISSUE #5942: node failed to start in simple_add_new_node_while_schema_changes_test
Seen in https://jenkins.scylladb.com/view/master/job/scylla-master/job/dtest-release/411/testReport/update_cluster_layout_tests/TestUpdateClusterLayout/simple_add_new_node_while_schema_changes_test/
Scylla version ba536dbc9530960a6132a0dc575fd31dee5365b2
(with repair-based ops)

https://jenkins.scylladb.com/view/master/job/scylla-master/job/dtest-release/411/artifact/logs-release.2/1583209437554_update_cluster_layout_tests.TestUpdateClusterLayout.simple_add_new_node_while_schema_changes_test/node4.log
```
INFO  2020-03-03 04:13:48,650 [shard 0] repair - repair id 1 on shard 0 failed: std::_Nested_exception<no_such_column_family> (Can't find a column family cf in keyspace ks): std::out_of_range ()
INFO  2020-03-03 04:13:48,650 [shard 0] repair - repair id 1 failed: std::runtime_error ({shard 0: std::_Nested_exception<no_such_column_family> (Can't find a column family cf in keyspace ks): std::out_of_range (), shard 1: std::_Nested_exception<no_such_column_family> (Can't find a column family cf in keyspace ks): std::out_of_range ()})
INFO  2020-03-03 04:13:48,650 [shard 0] repair - repair id 1 to sync data for keyspace=ks, status=failed: std::runtime_error ({shard 0: std::_Nested_exception<no_such_column_family> (Can't find a column family cf in keyspace ks): std::out_of_range (), shard 1: std::_Nested_exception<no_such_column_family> (Can't find a column family cf in keyspace ks): std::out_of_range ()})
...
INFO  2020-03-03 04:13:48,720 [shard 0] init - Startup failed: std::runtime_error ({shard 0: std::_Nested_exception<no_such_column_family> (Can't find a column family cf in keyspace ks): std::out_of_range (), shard 1: std::_Nested_exception<no_such_column_family> (Can't find a column family cf in keyspace ks): std::out_of_range ()})
```

### PR DESCRIPTION
**Pull request by asias**
repair: Ignore keyspace that is removed in sync_data_using_repair
    
When a keyspace is removed during node operations, we should not fail
the whole operation. Ignore the keyspace that is removed.

Fixes #5942

### CODE REVIEW

File: repair/repair.cc [hunk 1]
```diff
@@ -1559,7 +1559,11 @@ future<> sync_data_using_repair(seastar::sharded<database>& db,
             });
         }).then([id, keyspace] {
             rlogger.info("repair id {} to sync data for keyspace={}, status=succeeded", id, keyspace);
-        }).handle_exception([id, keyspace] (std::exception_ptr ep) {
+        }).handle_exception([&db, id, keyspace] (std::exception_ptr ep) {
+            if (!db.local().has_keyspace(keyspace)) {
+                rlogger.warn("repair id {} to sync data for keyspace={}, status=failed, keyspace does not exist any more, ignoring it: {}", id, keyspace, ep);
```
Comment (bhalevy) [inline repair/repair.cc]: nit: `s/failed,/failed:/`

Comment (asias) [reply repair/repair.cc]: There is the ":" in the end before the ep. However, I think it is better as you suggested. I pushed a new branch.

**Comment by asias** [comment]
Changes in 6a7c3f0af0bbf2d35c1dac7861ca3047e056a536:
- Put ":" after the word failed.

**Comment by asias** [comment]
ping

**Comment by bhalevy** [comment]
@nyh ping. please review.

**Pull request merged**